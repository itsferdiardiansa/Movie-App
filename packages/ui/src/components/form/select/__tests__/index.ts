import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeAll } from 'vitest'

import { Select } from '..'

describe('common/Form/Select', () => {
  let artist = ''
  let Component

  beforeAll(() => {
    const props = {
      placeholder: 'Please select',
      items: [
        { value: 'Muse', label: 'Muse' },
        { value: 'Alterbridge', label: 'Alterbridge' },
        { value: 'Lamb of God', label: 'Lamb of God' },
        { value: 'Erra', label: 'Erra' },
        { value: 'Architects', label: 'Architects' },
      ],
    }

    Component = mount(Select, {
      props: {
        modelValue: artist,
        ...props,
      },
    })
  })

  it('will render and match snapshot', () => {
    expect(Component.element).toMatchSnapshot()
  })
})
