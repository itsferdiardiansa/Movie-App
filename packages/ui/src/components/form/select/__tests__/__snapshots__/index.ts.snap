// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`common/Form/Select > will render and match snapshot 1`] = `
<div
  class="tx-control--input"
  data-v-896b2fa2=""
>
  <div
    class="tx-select"
    data-v-896b2fa2=""
  >
    <div
      class="tx-select--button"
      data-v-896b2fa2=""
      tabindex="0"
    >
      <span
        class="button-label"
        data-v-896b2fa2=""
      >
        Please select
      </span>
      <span
        class="button-icon"
        data-v-896b2fa2=""
      >
        <svg
          class="h-4"
          data-v-896b2fa2=""
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM377 271c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-87-87-87 87c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L239 167c9.4-9.4 24.6-9.4 33.9 0L377 271z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
    <transition-stub
      appear="true"
      css="true"
      data-v-896b2fa2=""
      name="slide-top"
      persisted="true"
    >
      <div
        class="tx-select--dropdown"
        data-v-896b2fa2=""
        style="display: none;"
      >
        <ul
          class="dropdown-list"
          data-v-896b2fa2=""
        >
          
          <li
            class="dropdown-item"
            data-v-896b2fa2=""
          >
            Muse
          </li>
          <li
            class="dropdown-item"
            data-v-896b2fa2=""
          >
            Alterbridge
          </li>
          <li
            class="dropdown-item"
            data-v-896b2fa2=""
          >
            Lamb of God
          </li>
          <li
            class="dropdown-item"
            data-v-896b2fa2=""
          >
            Erra
          </li>
          <li
            class="dropdown-item"
            data-v-896b2fa2=""
          >
            Architects
          </li>
          
        </ul>
      </div>
    </transition-stub>
  </div>
</div>
`;
